src/net/body.rs

```diff
@@ -357,12 +357,12 @@ enum FileField_ {
 }
 
 trait StreamField: Read + Send + 'static {
-    fn add_self(self: Box<Self>, name: String, filename: Option<String>, content_type: Option<Mime>, to: &mut Multipart);
+    fn add_self(self: Self, name: String, filename: Option<String>, content_type: Option<Mime>, to: &mut Multipart);
```

Comment:
On further investigation, it looks like this is hiding a double-boxing. See, the intent of taking `Box<Self>` was to force an unboxing before it re-boxed by adding itself to `Multipart`, which is basically doing the same thing as here. But because of the blanket impl matching the supertraits to `StreamField`, it turned out that `Box<StreamField>` also implemented `StreamField` itself, which is why this seems to work, but ultimately it's just double-boxing which is going to be detrimental to performance.

Feel free to revert this line and 364/365 or leave them, I'm going to be redoing this bit to avoid boxing at this level entirely.